docs/sphinx/examples/python/dynamics/ion_trap.py

```diff
@@ -0,0 +1,129 @@
+"""
+Tutorial: Creating a GHZ State with Trapped Ions
+
+This example shows how to prepare a GHZ (Greenberger-Horne-Zeilinger) state 
+using trapped ions and CUDA-Q. We'll use the effective spin model from the 
+famous Sørensen-Mølmer paper.
+
+What we're doing:
+- Start with N ions all in ground state |gg...g⟩
+- Apply an effective Hamiltonian H = 4χJ_x²  
+- Watch the system evolve into a GHZ superposition: (|gg...g⟩ + |ee...e⟩)/√2
+
+This tutorial uses the simplified effective model.
```

Comment:
Could you please add a reference to the derivation of the effective model (e.g., referring to a section/equation of the paper)?

This would help readers to understand the conversion from mathematical equations to CUDA-Q operator representation.